appinventor/components/src/com/google/appinventor/components/runtime/LocationSensor.java

```diff
@@ -84,8 +84,13 @@ public void onLocationChanged(Location location) {
         hasAltitude = true;
         altitude = location.getAltitude();
       }
-      hasLocationData = true;
-      LocationChanged(latitude, longitude, altitude);
+
+      // By default Location.latitude == Location.longitude == 0.
+      // So we want to ignore that case rather than generating a changed event.
+      if (longitude != UNKNOWN_VALUE && latitude != UNKNOWN_VALUE) {
```

Comment:
Yes, I think you are right -- i.e., as long as either lat or long is
non-zero it's a valid location.  Good catch.    I'll revise and resubmit.

Thanks,
-- ralph

On Tue, Dec 29, 2015 at 9:57 PM, hal <notifications@github.com> wrote:

> In
> appinventor/components/src/com/google/appinventor/components/runtime/LocationSensor.java
> <https://github.com/mit-cml/appinventor-sources/pull/646#discussion_r48584643>
> :
>
> > @@ -84,8 +84,13 @@ public void onLocationChanged(Location location) {
> >          hasAltitude = true;
> >          altitude = location.getAltitude();
> >        }
> > -      hasLocationData = true;
> > -      LocationChanged(latitude, longitude, altitude);
> > +
> > +      // By default Location.latitude == Location.longitude == 0.
> > +      // So we want to ignore that case rather than generating a changed event.
> > +      if (longitude != UNKNOWN_VALUE && latitude != UNKNOWN_VALUE) {
>
> should this be OR, not AND?
> what happens if the latitude is 0 and the longitude is non-zero?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/646/files#r48584643>.
>